cmd/genji/shell/command.go

```diff
@@ -366,3 +378,118 @@ func runDumpCmd(db *genji.DB, tables []string, w io.Writer) error {
 	_, err = fmt.Fprintln(w, "COMMIT;")
 	return err
 }
+
+// runSaveCommand saves the currently opened database in the given filename.
+// If a file already exists, it will be overwritten.
+func runSaveCmd(ctx context.Context, db *genji.DB, engineName string, path string) error {
+	tx, err := db.Begin(false)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	// Open the new database
+	var otherNg engine.Engine
+
+	switch engineName {
+	case "bolt":
+		otherNg, err = boltengine.NewEngine(path, 0660, nil)
+		if err != nil {
+			return err
+		}
+	case "badger":
+		otherNg, err = badgerengine.NewEngine(badger.DefaultOptions(path).WithLogger(nil))
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("Can't use unknown engine %s", engineName)
+	}
+
+	otherDB, err := genji.New(ctx, otherNg)
+	if err != nil {
+		return err
+	}
+	defer otherDB.Close()
+	otherTx, err := otherDB.Begin(true)
+	if err != nil {
+		return err
+	}
+	defer otherTx.Rollback()
+
+	// Find all tables
+	tables, err := tx.Query("SELECT table_name FROM __genji_tables")
+	if err != nil {
+		return err
+	}
+	defer tables.Close()
+
+	err = tables.Iterate(func(d document.Document) error {
+		// Get table name.
+		var tableName string
+		if err := document.Scan(d, &tableName); err != nil {
+			return err
+		}
+
+		table, err := tx.GetTable(tableName)
+		if err != nil {
+			return err
+		}
+
+		ti, err := table.Info()
+		if err != nil {
+			return err
+		}
+
+		otherTx.CreateTable(tableName, ti)
+		otherTable, err := otherTx.GetTable(tableName)
+		if err != nil {
+			return err
+		}
+
+		it := table.Store.Iterator(engine.IteratorOptions{})
+		defer it.Close()
+
+		var b []byte
+		for it.Seek(nil); it.Valid(); it.Next() {
+			itm := it.Item()
+			b, err := itm.ValueCopy(b)
+			if err != nil {
+				return err
+			}
+
+			otherTable.Store.Put(itm.Key(), b)
+		}
```

Comment:
after, check `it.Err()`

Reply:
Yes the `it.Err()` check is still missing, it must be done right outside of the for loop 👍🏼